subsystems/TypeHandlerLibrary/src/main/java/org/terasology/persistence/serializers/Serializer.java

```diff
@@ -22,6 +22,9 @@
  * methods that use {@link #serializeToPersisted(Object, TypeInfo)} and {@link #deserializeFromPersisted(PersistedData,
  * TypeInfo)}.
  */
+// log statements after an if are marked as false positive, suppress.
+// see pmd bug: https://github.com/pmd/pmd/issues/4731
+@SuppressWarnings("PMD.GuardLogStatementJavaUtil")
```

Comment:
is this about "[Logger calls should be surrounded by log level guards.](https://pmd.github.io/pmd-6.55.0/pmd_rules_java_bestpractices.html#guardlogstatement)" PMD findings?
I don't think they're false positives per se, as generally it can be good practise to have these guards.
However, in our case, I don't think the rule actually provides much benefit. we can discuss this in the reviver round, but I'd say that rule is a good candidate for removal instead of suppressing it everywhere.

Reply:
you were correct. there is an alternative: https://stackoverflow.com/questions/35096746/substituting-parameters-in-log-message-and-add-a-throwable-in-log4j-2, implemented.